metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/EntityResource.java

```diff
@@ -393,24 +399,78 @@ public Task<RollbackResponse> deleteEntities(@ActionParam("registryId") @Optiona
-      throws URISyntaxException {
+  public Task<DeleteEntityResponse> deleteEntity(@ActionParam(PARAM_URN) @Nonnull String urnStr,
+      @ActionParam(PARAM_ASPECT_NAME) @Nonnull String aspectName,
+      @ActionParam(PARAM_START_TIME_MILLIS) @Nullable Long startTimeMills,
+      @ActionParam(PARAM_END_TIME_MILLIS) @Nullable Long endTimeMillis) throws URISyntaxException {
     Urn urn = Urn.createFromString(urnStr);
     return RestliUtil.toTask(() -> {
       DeleteEntityResponse response = new DeleteEntityResponse();
 
       RollbackRunResult result = _entityService.deleteUrn(urn);
+      Long numTimeseriesDocsDeleted = deleteTimeseriesAspects(urn, startTimeMills, endTimeMillis, aspectName);
+      log.info("Total number of timeseries aspect docs deleted: {}", numTimeseriesDocsDeleted);
 
       response.setUrn(urnStr);
-      response.setRows(result.getRowsDeletedFromEntityDeletion());
+      response.setRows(result.getRowsDeletedFromEntityDeletion() + numTimeseriesDocsDeleted);
 
       return response;
     }, MetricRegistry.name(this.getClass(), "delete"));
   }
 
+  private Long deleteTimeseriesAspects(@Nonnull Urn urn, @Nullable Long startTimeMillis, @Nonnull Long endTimeMillis,
+      String aspectName) {
+    assert (urn != null);
+    long totalNumberOfDocsDeleted = 0;
+    // Construct the filter.
+    List<Criterion> criteria = new ArrayList<>();
+    Criterion hasUrnCriterion = new Criterion().setField("urn").setCondition(Condition.EQUAL).setValue(urn.toString());
+    criteria.add(hasUrnCriterion);
+
+    if (startTimeMillis != null) {
+      Criterion startTimeCriterion = new Criterion().setField(ES_FILED_TIMESTAMP)
+          .setCondition(Condition.GREATER_THAN_OR_EQUAL_TO)
+          .setValue(startTimeMillis.toString());
+      criteria.add(startTimeCriterion);
+    }
+    if (endTimeMillis != null) {
+      Criterion endTimeCriterion = new Criterion().setField(ES_FILED_TIMESTAMP)
+          .setCondition(Condition.LESS_THAN_OR_EQUAL_TO)
+          .setValue(endTimeMillis.toString());
+      criteria.add(endTimeCriterion);
+    }
+    Filter filter = QueryUtils.getFilterFromCriteria(criteria);
+
+    // Find the timeseries aspects to delete. If aspectName is null, delete all.
+    String entityType = urn.getEntityType();
+    EntitySpec entitySpec = _entityService.getEntityRegistry().getEntitySpec(entityType);
+    List<String> aspectsToDelete = (aspectName == null) ? entitySpec.getAspectSpecs()
+        .stream()
+        .filter(x -> x.isTimeseries())
+        .map(x -> x.getName())
+        .collect(Collectors.toList()) : entitySpec.getAspectSpecs()
+        .stream()
+        .filter(x -> x.isTimeseries() && x.getName().equals(aspectName))
+        .map(x -> x.getName())
+        .collect(Collectors.toList());
+
+    if (aspectName != null && aspectsToDelete.size() != 1) {
+      throw new IllegalArgumentException(String.format("Failed to find timeseries aspect {}", aspectName));
```

Comment:
We shouldn't throw an exception here. In the future, if we want to delete versioned aspect, this will always throw exception. I think it's cleaner to check whether the aspect is a timeseries outside of this function and do not call this function if the aspectName is not a timeseries (instead throw unsupported exception there)